pinot-common/src/main/java/org/apache/pinot/common/function/FunctionRegistry.java

```diff
@@ -147,4 +157,48 @@ public static FunctionInfo getFunctionInfo(String functionName, int numParameter
   private static String canonicalize(String functionName) {
     return StringUtils.remove(functionName, '_').toLowerCase();
   }
+
+  /**
+   * Placeholders for scalar function, they register and represents the signature for transform and filter predicate
+   * so that v2 engine can understand and plan them correctly.
+   */
+  private static class PlaceholderScalarFunctions {
+
+    public static Object jsonExtractScalar(String jsonFieldName, String jsonPath, String resultsType) {
+      throw new UnsupportedOperationException("Placeholder scalar function, should not reach here");
+    }
+
+    public static Object jsonExtractScalar(String jsonFieldName, String jsonPath, String resultsType,
+        Object defaultValue) {
+      throw new UnsupportedOperationException("Placeholder scalar function, should not reach here");
+    }
+
+    public static String jsonExtractKey(String jsonFieldName, String jsonPath) {
+      throw new UnsupportedOperationException("Placeholder scalar function, should not reach here");
+    }
+
+    public static long timeConvert(String timeConvertInput, String fromFormat, String toFormat) {
+      throw new UnsupportedOperationException("Placeholder scalar function, should not reach here");
+    }
+
+    public static long extract(String extractFormat, long timestamp) {
+      throw new UnsupportedOperationException("Placeholder scalar function, should not reach here");
+    }
+
+    // CHECKSTYLE:OFF
+    // @formatter:off
+    public static boolean text_contains(String text, String pattern) {
```

Comment:
Will calcite match the underscore? If so, other things can also break such as `json_extract_scalar`

Reply:
+1. will fix this by adding annotation but only register to calcite map